### ISSUE TITLE
train and val codes on Landsat-SCD dataset

### ISSUE THREAD

**Issue by syf0215** [ggsDing/SCanNet#6]
Hi, I'm very interested in your work. I have some problems when training on Landsat-SCD dataset. There are blank areas near the corner, and I dont know how to deal with it. I adopt the train and val codes in your previous work and just simply mask the input images and loss in blank areas. However, this leads to quite low SeK. 
Could you please give me a little comment or share your train and val codes on Landsat-SCD dataset? Thank you very much!

**Comment by YuhangZh** [comment]
I have the same problem, the metrics are worse when training the Landsat-SCD dataset.

**Comment by ggsDing** [comment]
Hi,
in [/datasets](https://github.com/ggsDing/SCanNet/tree/main/datasets) I have provided the pre-processing codes for the landsat-SCD dataset. Simply change the dataloader accordingly and adjust few arguments, then you can train on the landsat dataset.

**Comment by syf0215** [comment]
Thanks for your reply! I will try this.
Could you please tell me more about the training details of  **net_psd** in train_SCD_psd.py? Is it a trained teacher model? 
Without loading a trained model, the net_psd is a randomly initialized model and can't provide pseudo labels.
Should I first train a teacher model without pseudo labels, and then initialize the student model with the weights of teacher model, so that the net_psd can work as the teacher model in teacher-student paradigm?

**Comment by ggsDing** [comment]
> Thanks for your reply! I will try this. Could you please tell me more about the training details of **net_psd** in train_SCD_psd.py? Is it a trained teacher model? Without loading a trained model, the net_psd is a randomly initialized model and can't provide pseudo labels. Should I first train a teacher model without pseudo labels, and then initialize the student model with the weights of teacher model, so that the net_psd can work as the teacher model in teacher-student paradigm?

Hi,
You can check line 79 and line 228 in the [training code](https://github.com/ggsDing/SCanNet/blob/main/train_SCD_psd.py). Net_psd directly copy the previous round of trained weights, if its accuracy is above a certain level.

**Comment by ggsDing** [comment]
> I have the same problem, the metrics are worse when training the Landsat-SCD dataset.

The pre-processed data have been updated in the repository description. Please check :)

**Comment by ggsDing** [comment]
> Hi, I'm very interested in your work. I have some problems when training on Landsat-SCD dataset. There are blank areas near the corner, and I dont know how to deal with it. I adopt the train and val codes in your previous work and just simply mask the input images and loss in blank areas. However, this leads to quite low SeK. Could you please give me a little comment or share your train and val codes on Landsat-SCD dataset? Thank you very much!

The pre-processed data have been updated in the repository description. Please check :)

**Comment by Mr-wang27** [comment]
@YuhangZh @syf0215 @DingLei14 
 Hello, could you please tell me how you trained the Landsat-SCD dataset? I used Professor Ding Lei's training code, but encountered NaN during the first epoch. Could you advise me on how to train this dataset? Thank you very much for your help!
